src/fu-util.c

```diff
@@ -237,6 +238,26 @@ fu_util_prompt_for_number (guint maxnum)
 	return answer;
 }
 
+static gboolean
+fu_util_prompt_for_boolean (gboolean def)
+{
+	do {
+		char buffer[4];
+		if (!fgets (buffer, sizeof (buffer), stdin))
+			continue;
+		if (strlen (buffer) == sizeof (buffer) - 1)
+			continue;
+		if (g_strcmp0 (buffer, "\n") == 0)
+			return def;
+		buffer[0] = g_ascii_toupper (buffer[0]);
+		if (g_strcmp0 (buffer, "Y\n") == 0)
```

Comment:
buffer cant be NULL, you could omit the additionally null-check of `g_strcmp0`.

Reply:
Valid, but not performance critical.